### PR TITLE
feat: Replace icons with uploaded profile picture

### LINKED ISSUE #10251: Replace icons with uploaded profile picture
After the name and profile picture are specified, lets show it as icon and name in the wallet sidebar 

image

- in the navbar 

image

- in the wallet sidebar

----------
as specified in here: 
  
image

(showing funds for multiple funds in the wallet sidebar will have to be figured out - note for me)

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Feature

## Needs Design check

- @exezbcz please review

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #10251

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image

### CODE REVIEW

**Comment by prury** [comment]
@Jarsen136 from what i could verify, the test is interrupted because the following error is thrown when accessing a non-new-structured profile: 

Error: Failing test due to error in browser: FetchError: [GET] "https://profile-beta.kodadot.workers.dev/profiles/5Evgu7n2pVLkj6QfAdDAx11yqmQKstAJbkPCnscYf9pMxTja": 404 

is there a way to handle it?

**Comment by prury** [comment]
i think we may improve it a bit and not show the old wallet image when opening the profile, maybe a skeleton? @exezbcz wdyt


[Gravação de tela de 07-05-2024 13:47:48.webm](https://github.com/kodadot/nft-gallery/assets/36627808/af9fdee0-be63-4a02-80d5-45e8aca221ca)

File: components/identity/IdentityItem.vue [hunk 1]
```diff
@@ -33,6 +34,7 @@ withDefaults(
     prefix: string
     hideIdentityPopover?: boolean
     disableIdentityLink?: boolean
+    showProfileAvatar?: boolean
```
Comment (daiagi) [inline components/identity/IdentityItem.vue:37]: missing default value?

**Comment by exezbcz** [comment]
amazing! Looking fresh


could you also please change the "address" to the offchain name

image


i will make some ux improvements followups to ensure it does not compromise the usability

**Review by daiagi — APPROVED** [review]

**Comment by exezbcz** [comment]
@prury hmm interesting that its not in the cache?

File: components/identity/IdentityItem.vue (hunk 1 above)
Comment (Jarsen136) [reply components/identity/IdentityItem.vue]: updated

**Comment by Jarsen136** [comment]
> amazing! Looking fresh
> 
> could you also please change the "address" to the offchain name
> 

✅  Updated
image

**Comment by Jarsen136** [comment]
> i think we may improve it a bit and not show the old wallet image when opening the profile, maybe a skeleton? @exezbcz wdyt
> 

> @prury hmm interesting that its not in the cache?

I have added a better cache for the profile. Could you please test it again?

**Comment by prury** [comment]
> > i think we may improve it a bit and not show the old wallet image when opening the profile, maybe a skeleton? @exezbcz wdyt
> 
> > @prury hmm interesting that its not in the cache?
> 
> I have added a better cache for the profile. Could you please test it again?

much better thank you!

File: composables/useFetchProfile.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+export default function useFetchProfile(address: Ref<string | undefined>) {
+  const { data: profile } = useAsyncData<Profile | null>(
+    `userProfile-${address.value}`,
+    () => {
+      if (address.value) {
+        return fetchProfileByAddress(address.value)
+      }
+      return Promise.resolve(null)
+    },
```
Comment (hassnian) [inline composables/useFetchProfile.ts:12]:
```suggestion
    () =>  address.value ? fetchProfileByAddress(address.value) : Promise.resolve(null),
```

**Review by hassnian — APPROVED** [review]
otherwise LGTM

**Pull request merged**